openmdao/docs/features/building_blocks/components/mux_demux_comps.rst

```diff
@@ -0,0 +1,74 @@
+
+.. _mux_demux_comp_feature:
+
+*********************
+MuxComp and DemuxComp
+*********************
+
+`DemuxComp` and `MuxComp` work together to break up inputs into multiple values (demux) or combine
+multiple inputs into a single value (mux).  This can be useful in situations where scalar outputs
+from multiple components need to be fed into a single vectorized component.
+
+`DemuxComp` takes a single input of arbitary shape (the size of at least one axis must be equal
+to `vec_size`).  In can then be broken along that axis, resulting in `vec_size` outputs.
+
+`MuxComp` combines two or more inputs into a single output by stacking them along an axis.
+
+MuxComp and DemuxComp Options
+-----------------------------
+
+These components have a single option, `vec_size`, which provides the number of inputs to be
+combined into a single output (for `MuxComp`) or the number of outputs into which an input is
+to be broken (for `DemuxComp`).  The default value of `vec_size` is 2.
+
+Adding Variables
+----------------
+
+A single `MuxComp` or `DemuxComp` can mux or demux multiple variables, so long as all variables
+are compatible with the given `vec_size`.  Variables are added via the `add_var` method.
+
+The axis along which the muxing/demuxing is to occur is given via the axis argument.  For DemuxComp,
+the specified axis must be be less than or equal to the number of dimensions in shape,
+otherwise an exception is raised.  In addition, the axis on which the Demuxing is to be done must
+have length `vec_size`.  For MuxComp, a valid stacking axis is less than *ndim* + 1.
+
+For DemuxComp, the name of the given variable is the **input**.  It is demuxed into variables whose
+names are appended with `_n` where `n` is an integer from 0 through `vec_size`-1.  Conversely, for
+MuxComp, the given variable name is the output, and each input is appended with `_n`.
+
+.. py:method:: add_var(self, name, val=1.0, shape=None, units=None, desc='', axis=0):
+
+    Add a variable to be muxed or demuxed, and all associated input/output variables
+
```

Comment:
Hmm, don't we have a way to embed a method signature and docstring? @channel

Reply:
I think you're talking about automethod, documented here:

http://openmdao.org/twodocs/versions/latest/developer_docs/style_guide/doc_style_guide.html#embedding-autodocumentation-snippets-into-documentation